vendor/github.com/containers/storage/pkg/idtools/idtools_unix.go

```diff
@@ -26,10 +26,14 @@ func mkdirAs(path string, mode os.FileMode, ownerUID, ownerGID int, mkAll, chown
 	// so that we can chown all of them properly at the end.  If chownExisting is false, we won't
 	// chown the full directory path if it exists
 	var paths []string
+	canChown := os.Getuid() == 0
```

Comment:
All of this also seems like a decision that should be visible at the caller level, of possibly a few levels higher.